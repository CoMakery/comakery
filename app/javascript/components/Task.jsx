import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import Icon from './styleguide/Icon'
import CurrencyAmount from './CurrencyAmount'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.1);
  height: 60px;
  margin-bottom: 10px;
`

const RightBorder = styled.div`
  width: 2px;
  height: 60px;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, .2);
  background-color: #5037f7;
  z-index: 10;
  position: absolute;
  margin-left: -1px;

  ${props => props.status === 'ready' && css`
    background-color: #5037f7;
  `}

  ${props => props.status === 'accepted' && css`
    background-color: #7ed321;
  `}

  ${props => props.status === 'paid' && css`
    background-color: #fb40e5;
  `}
`

const Info = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`

const Name = styled.div`
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  padding-bottom: 8px;
`

const Status = styled.div`
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-transform: uppercase;

  ${props => props.status === 'ready' && css`
    color: #5037f7;
  `}

  ${props => props.status === 'accepted' && css`
    color: #7ed321;
  `}

  ${props => props.status === 'paid' && css`
    color: #fb40e5;
  `}
`

const Title = styled.div`
  padding: 7px 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`

const Details = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: space-between;
`

const Amount = styled(CurrencyAmount)`
`

const Buttons = styled.div`
  display: flex;
  justify-content: space-between;
  margin-right: 15px;
  align-items: center;
  padding-top: 8px;
`

const PaymentButton = styled.a`
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #0089f4;
  text-decoration: none;
  text-transform: uppercase;

  &:hover,
  &:focus {
    text-decoration: underline;
  }
`

const StyledIcon = styled(Icon)`
  height: 15px;
  margin-left: 16px;
`

class Task extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Wrapper>
          <RightBorder status={this.props.task.status} />
          <Info>
            <Title>
              <Name>
                {this.props.task.name}
              </Name>

              <Status status={this.props.task.status}>
                {this.props.task.status}
              </Status>
            </Title>

            <Details>
              <Amount
                amount={this.props.task.totalAmount}
                currency={this.props.task.currency}
                logoUrl={this.props.task.currencyLogo}
              />

              <Buttons>
                {this.props.task.status === 'ready' &&
                <PaymentButton href={this.props.task.awardPath}>
                    issue award
                </PaymentButton>
                }
                {this.props.task.status === 'accepted' &&
                <PaymentButton href={this.props.task.payPath}>
                    pay contributor
                </PaymentButton>
                }
                <a href={this.props.task.editPath}>
                  <StyledIcon name="iconEdit.svg" />
                </a>
                <a rel="nofollow" data-method="delete" href={this.props.task.destroyPath}>
                  <StyledIcon name="iconTrash.svg" />
                </a>
              </Buttons>
            </Details>
          </Info>
        </Wrapper>
      </React.Fragment>
    )
  }
}

Task.propTypes = {
  task: PropTypes.object
}
Task.defaultProps = {
  task: {
    status: null
  }
}
export default Task
