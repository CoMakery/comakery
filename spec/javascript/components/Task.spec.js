import React from 'react'
import { mount } from 'enzyme'
import Task from 'components/Task'

describe('Task', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<Task />)

    expect(wrapper).toMatchSnapshot()
  })
})
