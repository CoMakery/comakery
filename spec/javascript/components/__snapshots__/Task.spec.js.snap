// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task renders component correctly 1`] = `
<Task
  editable={true}
  task={
    Object {
      "contributor": Object {
        "image": null,
        "name": null,
      },
      "status": null,
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gZMcBi",
            "isStatic": true,
            "lastClassName": "ivguJA",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.1);
  height: 85px;
  margin-bottom: 10px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gZMcBi",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-gZMcBi ivguJA"
      >
        <styled.div
          status={null}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gqjmRU",
                  "isStatic": false,
                  "lastClassName": "dHwgxQ",
                  "rules": Array [
                    "
  width: 2px;
  height: 85px;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, .2);
  background-color: #5037f7;
  z-index: 10;
  position: absolute;
  margin-left: -1px;

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-gqjmRU",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            status={null}
          >
            <div
              className="sc-gqjmRU dHwgxQ"
            />
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-VigVT",
                  "isStatic": true,
                  "lastClassName": "fvfiCJ",
                  "rules": Array [
                    "
  display: flex;
  flex-direction: column;
  width: 100%;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-VigVT",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-VigVT fvfiCJ"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-chPdSV",
                        "isStatic": true,
                        "lastClassName": "jPGuix",
                        "rules": Array [
                          "
  padding: 7px 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-chPdSV",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-chPdSV jPGuix"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-jTzLTM",
                              "isStatic": true,
                              "lastClassName": "CyBoy",
                              "rules": Array [
                                "
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  padding-bottom: 8px;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-jTzLTM",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-jTzLTM CyBoy"
                        />
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-kgoBCf",
                        "isStatic": true,
                        "lastClassName": "gZLJew",
                        "rules": Array [
                          "
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: space-between;
  padding-right: 15px;
  padding-top: 7px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-kgoBCf",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-kgoBCf gZLJew"
                  >
                    <Styled(CurrencyAmount)>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-kGXeez",
                              "isStatic": true,
                              "lastClassName": "fqVwMj",
                              "rules": Array [
                                "
",
                              ],
                            },
                            "displayName": "Styled(CurrencyAmount)",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-kGXeez",
                            "target": [Function],
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <CurrencyAmount
                          amount=""
                          className="sc-kGXeez fqVwMj"
                          currency={null}
                          logoUrl=""
                        >
                          <styled.div>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-ifAKCX",
                                    "isStatic": true,
                                    "lastClassName": "fOIRMD",
                                    "rules": Array [
                                      "
  display: flex;
  flex-direction: row;
  align-items: center;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 500;
  color: #4a4a4a;
",
                                    ],
                                  },
                                  "displayName": "styled.div",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-ifAKCX",
                                  "target": "div",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <div
                                className="sc-ifAKCX fOIRMD"
                              >
                                <styled.div>
                                  <StyledComponent
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "sc-EHOje",
                                          "isStatic": true,
                                          "lastClassName": "iqtoiU",
                                          "rules": Array [
                                            "
  margin-right: 0.5em;
",
                                          ],
                                        },
                                        "displayName": "styled.div",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "sc-EHOje",
                                        "target": "div",
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                  >
                                    <div
                                      className="sc-EHOje iqtoiU"
                                    />
                                  </StyledComponent>
                                </styled.div>
                              </div>
                            </StyledComponent>
                          </styled.div>
                        </CurrencyAmount>
                      </StyledComponent>
                    </Styled(CurrencyAmount)>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-kpOJdX",
                              "isStatic": true,
                              "lastClassName": "knRMxI",
                              "rules": Array [
                                "
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-top: 8px;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-kpOJdX",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-kpOJdX knRMxI"
                        >
                          <Styled(Icon)
                            name="INVITE_USER.svg"
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-jKJlTe",
                                    "isStatic": true,
                                    "lastClassName": "IiUff",
                                    "rules": Array [
                                      "
  height: 15px;
  width: 15px;
  margin-left: 16px;
  opacity: 0.3;
  cursor: not-allowed;
",
                                    ],
                                  },
                                  "displayName": "Styled(Icon)",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-jKJlTe",
                                  "target": [Function],
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              name="INVITE_USER.svg"
                            >
                              <Icon
                                className="sc-jKJlTe IiUff"
                                name="INVITE_USER.svg"
                              >
                                <img
                                  className="icon icon__INVITE_USER-svg sc-jKJlTe IiUff"
                                  src="test-file-stub"
                                />
                              </Icon>
                            </StyledComponent>
                          </Styled(Icon)>
                          <a>
                            <Styled(Icon)
                              name="DUPLICATE.svg"
                            >
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-ckVGcZ",
                                      "isStatic": true,
                                      "lastClassName": "bzzJJm",
                                      "rules": Array [
                                        "
  height: 15px;
  width: 15px;
  margin-left: 16px;
",
                                      ],
                                    },
                                    "displayName": "Styled(Icon)",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-ckVGcZ",
                                    "target": [Function],
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                                name="DUPLICATE.svg"
                              >
                                <Icon
                                  className="sc-ckVGcZ bzzJJm"
                                  name="DUPLICATE.svg"
                                >
                                  <img
                                    className="icon icon__DUPLICATE-svg sc-ckVGcZ bzzJJm"
                                    src="test-file-stub"
                                  />
                                </Icon>
                              </StyledComponent>
                            </Styled(Icon)>
                          </a>
                          <Styled(Icon)
                            name="iconEdit.svg"
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-jKJlTe",
                                    "isStatic": true,
                                    "lastClassName": "IiUff",
                                    "rules": Array [
                                      "
  height: 15px;
  width: 15px;
  margin-left: 16px;
  opacity: 0.3;
  cursor: not-allowed;
",
                                    ],
                                  },
                                  "displayName": "Styled(Icon)",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-jKJlTe",
                                  "target": [Function],
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              name="iconEdit.svg"
                            >
                              <Icon
                                className="sc-jKJlTe IiUff"
                                name="iconEdit.svg"
                              >
                                <img
                                  className="icon icon__iconEdit-svg sc-jKJlTe IiUff"
                                  src="test-file-stub"
                                />
                              </Icon>
                            </StyledComponent>
                          </Styled(Icon)>
                          <Styled(Icon)
                            name="iconTrash.svg"
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-jKJlTe",
                                    "isStatic": true,
                                    "lastClassName": "IiUff",
                                    "rules": Array [
                                      "
  height: 15px;
  width: 15px;
  margin-left: 16px;
  opacity: 0.3;
  cursor: not-allowed;
",
                                    ],
                                  },
                                  "displayName": "Styled(Icon)",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-jKJlTe",
                                  "target": [Function],
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              name="iconTrash.svg"
                            >
                              <Icon
                                className="sc-jKJlTe IiUff"
                                name="iconTrash.svg"
                              >
                                <img
                                  className="icon icon__iconTrash-svg sc-jKJlTe IiUff"
                                  src="test-file-stub"
                                />
                              </Icon>
                            </StyledComponent>
                          </Styled(Icon)>
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-kEYyzF",
                  "isStatic": true,
                  "lastClassName": "jOdmLg",
                  "rules": Array [
                    "
  display: flex;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-kEYyzF",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-kEYyzF jOdmLg"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-fjdhpX",
                        "isStatic": true,
                        "lastClassName": "fkDFwz",
                        "rules": Array [
                          "
  display: flex;
  padding: 7px 20px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-fjdhpX",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-fjdhpX fkDFwz"
                  >
                    <styled.div
                      status={null}
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-jzJRlG",
                              "isStatic": false,
                              "lastClassName": "cwLuvr",
                              "rules": Array [
                                "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  margin-right: 2em;

  ",
                                [Function],
                                "

  ",
                                [Function],
                                "

  ",
                                [Function],
                                "

  ",
                                [Function],
                                "

  ",
                                [Function],
                                "

  ",
                                [Function],
                                "

  ",
                                [Function],
                                "
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-jzJRlG",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        status={null}
                      >
                        <div
                          className="sc-jzJRlG cwLuvr"
                        />
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.a
                onClick={[Function]}
                rotateIcon={false}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-eNQAEJ",
                        "isStatic": false,
                        "lastClassName": "gnKmFP",
                        "rules": Array [
                          "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #201662;
  margin-left: auto;
  text-transform: uppercase;
  text-decoration: none;
  display: flex;
  flex-direction: row;
  align-items: center;
  cursor: pointer;
  margin: .5em 1.5em .5em auto;

  &:hover {
    text-decoration: underline;
  }

  img {
    margin-left: 7px;
    ",
                          [Function],
                          "
  }

  @media (max-width: 1024px) {
    margin-left: initial;
    margin-top: 15px;
    margin-bottom: 15px;
  }
",
                        ],
                      },
                      "displayName": "styled.a",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-eNQAEJ",
                      "target": "a",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                  rotateIcon={false}
                >
                  <a
                    className="sc-eNQAEJ gnKmFP"
                    onClick={[Function]}
                  >
                    Details 
                    <Icon
                      className=""
                      name="DROP_DOWN.svg"
                    >
                      <img
                        className="icon icon__DROP_DOWN-svg"
                        src="test-file-stub"
                      />
                    </Icon>
                  </a>
                </StyledComponent>
              </styled.a>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Task>
`;
